pkg/plugin/processor/builtin/impl/webhook/http.go

```diff
@@ -72,6 +75,32 @@ type httpConfig struct {
 	ResponseStatusRef string `json:"response.status"`
 }
 
+func (c *httpConfig) parseHeaders() error {
+	if c.Headers == nil {
+		c.Headers = make(map[string]string)
+	}
+
+	var hName, hValue string
+	for name, value := range c.Headers {
+		if strings.ToLower(name) == "content-type" {
+			hValue = value
+			hName = name
+		}
+	}
+
+	if hValue != "" && c.ContentType != "" {
+		return cerrors.Errorf("Configuration error, cannot provide both \"request.contentType\" and \"headers.Content-Type\", use \"headers.Content-Type\" only.")
+	}
+
+	delete(c.Headers, hName)
+	c.Headers["Content-Type"] = cmp.Or(hValue, c.ContentType)
```

Comment:
I'll say, there's an edge case, a bit exotic, but still. If I provide two headers, `content-type` and `Content-Type`, it's random if one or both headers will be sent by the processor, depending on how Go decides to traverse the `c.Headers` map.

My proposal, to make this completely safe:
```suggestion
	if c.ContentType == "" {
		return nil // Nothing to replace in headers
	}

	for name, _ := range c.Headers {
		if strings.ToLower(name) == "content-type" {
			return cerrors.Errorf("Configuration error, cannot provide both \"request.contentType\" and \"headers.Content-Type\", use \"headers.Content-Type\" only.")
		}
	}

	c.Headers["Content-Type"] = c.ContentType
```